File: sample/JumioMobileSample/src/main/java/com/jumio/sample/java/netverify/customui/BottomSheetDialogDocuments.java
package com.jumio.sample.java.netverify.customui;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.jumio.nv.data.document.NVDocumentType;
import com.jumio.nv.data.document.NVDocumentVariant;
import com.jumio.sample.R;

import java.util.ArrayList;
import java.util.List;

public class BottomSheetDialogDocuments extends BottomSheetDialogFragment implements View.OnClickListener {

	private List<LinearLayout> optionLayouts = new ArrayList<>();
	private List<ImageView> imageViews = new ArrayList<>();
	private List<TextView> textViews = new ArrayList<>();
	private List<NVDocumentType> documentTypes = new ArrayList<>();
	private OnBottomSheetActionListener callback;

	public interface OnBottomSheetActionListener {
		void onDocumentTypeSelected(NVDocumentType type, NVDocumentVariant variant, boolean variantSelected);
	}

	public BottomSheetDialogDocuments() {
		// Required empty public constructor
	}

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		Bundle bundle = getArguments();
		if (bundle != null) {
			documentTypes = new ArrayList<NVDocumentType>() ;
			ArrayList<String> stringList = bundle.getStringArrayList(NetverifyCustomActivity.BUNDLE_DOCUMENT_TYPE_LIST);
			if(stringList != null) {
				for(String s : stringList) {
					documentTypes.add(NVDocumentType.fromString(s));
				}
			}

		}

		//initialize callback
		callback = (OnBottomSheetActionListener) getActivity();
	}

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container,
	                         Bundle savedInstanceState) {
		View root = inflater.inflate(R.layout.bottom_sheet_dialog, container, false);

		optionLayouts.add(root.findViewById(R.id.bottom_sheet_documents_option1));
		optionLayouts.add(root.findViewById(R.id.bottom_sheet_documents_option2));
		optionLayouts.add(root.findViewById(R.id.bottom_sheet_documents_option3));
		optionLayouts.add(root.findViewById(R.id.bottom_sheet_documents_option4));

		for (LinearLayout l : optionLayouts) {
			l.setOnClickListener(this);
		}

		imageViews.add(root.findViewById(R.id.bottom_sheet_documents_option1_image));
		imageViews.add(root.findViewById(R.id.bottom_sheet_documents_option2_image));
		imageViews.add(root.findViewById(R.id.bottom_sheet_documents_option3_image));
		imageViews.add(root.findViewById(R.id.bottom_sheet_documents_option4_image));

		textViews.add(root.findViewById(R.id.bottom_sheet_documents_option1_text));
		textViews.add(root.findViewById(R.id.bottom_sheet_documents_option2_text));
		textViews.add(root.findViewById(R.id.bottom_sheet_documents_option3_text));
		textViews.add(root.findViewById(R.id.bottom_sheet_documents_option4_text));

		int i = 0;
		for (NVDocumentType type : documentTypes) {
			if (i < documentTypes.size()) {
				optionLayouts.get(i).setVisibility(View.VISIBLE);
				imageViews.get(i).setImageResource(getIconForDocumentType(type));
				textViews.get(i).setText(type.getLocalizedName(getActivity()));
				i++;
			}
		}

		return root;
	}

	private int getIconForDocumentType(NVDocumentType documentType) {
		switch (documentType) {
			case DRIVER_LICENSE:
				return R.drawable.ic_selection_driver_license;
			case VISA:
				return R.drawable.ic_selection_visa;
			case PASSPORT:
				return R.drawable.ic_selection_passport;
			case IDENTITY_CARD:
				return R.drawable.ic_selection_id_card;
		}
		return 0;
	}

	@Override
	public void onClick(View v) {
		if (v == null) {
			return;
		}
		callback.onDocumentTypeSelected(documentTypes.get(optionLayouts.indexOf(v)), NVDocumentVariant.PLASTIC, false);
		dismiss();
	}
}

File: sample/JumioMobileSample/src/main/java/com/jumio/sample/kotlin/netverify/customui/BottomSheetDialogDocuments.kt
package com.jumio.sample.kotlin.netverify.customui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.jumio.nv.data.document.NVDocumentType
import com.jumio.nv.data.document.NVDocumentVariant
import com.jumio.sample.R
import java.util.*

class BottomSheetDialogDocuments : BottomSheetDialogFragment(), View.OnClickListener {
	private val optionLayouts: MutableList<LinearLayout> = ArrayList()
	private val imageViews: MutableList<ImageView> = ArrayList()
	private val textViews: MutableList<TextView> = ArrayList()
	private var documentTypes: List<NVDocumentType>? = ArrayList()
	private var callback: OnBottomSheetActionListener? = null

	interface OnBottomSheetActionListener {
		fun onDocumentTypeSelected(documentType: NVDocumentType?, documentVariant: NVDocumentVariant?, variantSelected: Boolean?)
	}

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		val bundle = arguments
		if (bundle != null) {
			documentTypes = bundle.getStringArrayList(NetverifyCustomActivity.BUNDLE_DOCUMENT_TYPE_LIST)?.map { NVDocumentType.fromString(it) }
		}
		//initialize callback
		callback = activity as OnBottomSheetActionListener?
	}

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
	                          savedInstanceState: Bundle?): View? {
		val root = inflater.inflate(R.layout.bottom_sheet_dialog, container, false)
		optionLayouts.add(root.findViewById(R.id.bottom_sheet_documents_option1))
		optionLayouts.add(root.findViewById(R.id.bottom_sheet_documents_option2))
		optionLayouts.add(root.findViewById(R.id.bottom_sheet_documents_option3))
		optionLayouts.add(root.findViewById(R.id.bottom_sheet_documents_option4))
		for (l in optionLayouts) {
			l.visibility = View.GONE
			l.setOnClickListener(this)
		}
		imageViews.add(root.findViewById(R.id.bottom_sheet_documents_option1_image))
		imageViews.add(root.findViewById(R.id.bottom_sheet_documents_option2_image))
		imageViews.add(root.findViewById(R.id.bottom_sheet_documents_option3_image))
		imageViews.add(root.findViewById(R.id.bottom_sheet_documents_option4_image))
		textViews.add(root.findViewById(R.id.bottom_sheet_documents_option1_text))
		textViews.add(root.findViewById(R.id.bottom_sheet_documents_option2_text))
		textViews.add(root.findViewById(R.id.bottom_sheet_documents_option3_text))
		textViews.add(root.findViewById(R.id.bottom_sheet_documents_option4_text))
		var i = 0
		for (type in documentTypes!!) {
			if (i < documentTypes!!.size) {
				optionLayouts[i].visibility = View.VISIBLE
				imageViews[i].setImageResource(getIconForDocumentType(type))
				textViews[i].text = type.getLocalizedName(activity)
				i++
			}
		}
		return root
	}

	private fun getIconForDocumentType(documentType: NVDocumentType?): Int {
		return when (documentType) {
			NVDocumentType.DRIVER_LICENSE -> R.drawable.ic_selection_driver_license
			NVDocumentType.VISA -> R.drawable.ic_selection_visa
			NVDocumentType.PASSPORT -> R.drawable.ic_selection_passport
			NVDocumentType.IDENTITY_CARD -> R.drawable.ic_selection_id_card
			else -> return 0
		}
	}

	override fun onClick(v: View?) {
		if (v == null) {
			return
		}
		callback!!.onDocumentTypeSelected(documentTypes!![optionLayouts.indexOf(v)], NVDocumentVariant.PLASTIC, false)
		dismiss()
	}
}